RPM/Tags.hs

```diff
@@ -682,52 +682,60 @@ mkNull _ ty _ _ | ty == 0    = Just Null
                 | otherwise  = Nothing
 
 mkChar :: BS.ByteString -> Word32 -> Word32 -> Word32 -> Maybe [Char]
-mkChar store ty offset count | ty == 1   = Just $ C.unpack $ BS.take count' start
+mkChar store ty offset count | fromIntegral (BS.length store) < count = Nothing
+                             | ty == 1   = Just $ C.unpack $ BS.take count' start
                              | otherwise = Nothing
```

Comment:
I think you need to take `offset` into account here when doing the size comparison.  `store` is the entirety of the header store - it contains the values not just for this tag, but for all other tags in the RPM.  `offset` is where in the store to start reading for this tag.  Other tests will need to be updated similarly.